Added delete _progress function

Cheethan opened the pull request:
Fixes #1 

- Added delete _progress that clears screen of Soil,Plants ( along with growth_stage ) and Water 

Let me know if you'd like any changes

BDadmehr0 reviewed (changes requested):
Make it so that blocks can be deleted one by one (in the same way they are added), not all at once.

Cheethan commented:
> Make it so that blocks can be deleted one by one (in the same way they are added), not all at once.

- Added tile deletion bound to D
- Renamed delete_progress() to reset_map() — it's now bound to R.

Let me know if any changes are to be made.

BDadmehr0 reviewed (approved):
LGTM! @Cheethan
